docs/src/architecture/passwordless_authentication.md

```diff
@@ -0,0 +1,41 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Yubikey Passwordless Authentication
+
+This section describes Ghaf reference implementation for passwordless authentication for fast identity online (FIDO). The reference implementation has been created with Yubico Yubikeys.
+The implementation is modular and configurable - thus enabling also other implementations.
+
+This section describes Yubikey Passwordless Authentication and how to create U2F keys.
+
+The reader is expected to know the fundamentals of [FIDO](https://fidoalliance.org/specifications/), [Yubico](https://www.yubico.com/) and [U2F](https://developers.yubico.com/U2F/).
+
+## Prerequisites for Yubikey Passwordless Authentication
+
+User must have compatible [YubiKey hardware](https://www.yubico.com/products/) to begin with.
+
+## Creating Yubikey U2F Keys
+
+Yubikey U2F keys can be created with [pamu2fcfg](https://developers.yubico.com/pam-u2f/), a module implements PAM over U2F and FIDO2. pamu2fcfg is available in Nixpkgs as pkgs.pamu2fcfg.
+
+Use the following command to create your U2F keys for Ghaf gui-vm .
+```
+$ pamu2fcfg -u ghaf -o pam://gui-vm | sudo tee /etc/u2f_mappings
+```
+
+After running above command, there will be green light blinking on Yubikey and user need to touch it.
+Once executed successfully, now user authentication becomes passwordless. When prompted with password just touch Yubikey.
+More details about mapping file can found here [central-authorization-mapping](https://github.com/Yubico/pam-u2f#central-authorization-mapping)
+
+## Current Implementation
+
+Since passwordless authentication is mapped with Yubikey in use and each Yubikey will generate unique key. So its not possible to use generic key.
+Yubikey Passwordless Authentication feature is enabled in debug builds for Ghaf gui-vm only.
+We are using pre-generated public keys which are created using pam2fcfg tool on different machine and it work automatically on guivm after the public key part is included in the built image for specific Yubikey.
```

Comment:
Could we add:
...guivm after the public key part is included in the built image for specific Yubikey by running
`pamu2fcfg -u ghaf -o pam://gui-vm | sudo tee ./modules/host/demo-yubikeys/u2f_mappings`
withing the ghaf repository before building.

Reply:
Something like that might be useful improvement for the initial key generation (clean Yubikey).
There's also the scenario where the user already has Yubikey (with private key in it) and wants to only add the existing public key (which matches the private key in the Yubikey) to the authorized keys (`u2f_mappings`).